### PR TITLE
Added test cases for validate_overlap function and added docstrings

### PR DESCRIPTION
**Pull request by swortelboer**
- Added test cases for the validate_overlap function in membership.py.
- Included simple docstring explanations for each test case.
- Fixed a typo in the validate_overlap function.

### CODE REVIEW

File: loefsys/contacts/tests/test_models.py [hunk 1]
```diff
@@ -42,42 +54,206 @@ def test_validation_person(self):
...
+            member=self.member,
+            start=date(2023, 1, 1),
+            end=date(2023, 12, 31),
+        )
+        self.assertFalse(validate_overlap(membership1, [membership2]))
+
+    def test_overlap(self):
+        """Test that there is overlap when the membership dates overlap."""
+        membership1 = G(
+            Membership, member=self.member, start=date(2023, 6, 1), end=date(2024, 6, 1)
```
Comment (SypTitan) [inline loefsys/contacts/tests/test_models.py:139]:
```suggestion
            Membership, 
            member=self.member, 
            start=date(2023, 6, 1), 
            end=date(2024, 6, 1)
```

File: loefsys/contacts/tests/test_models.py [hunk 2]
```diff
@@ -42,42 +54,206 @@ def test_validation_person(self):
...
+            member=self.member,
+            start=date(2023, 1, 1),
+            end=date(2023, 12, 31),
+        )
+        self.assertTrue(validate_overlap(membership1, [membership2]))
+
+    def test_overlap_same_start(self):
+        """Test that there is overlap when the membership has the same start date."""
+        membership1 = G(
+            Membership, member=self.member, start=date(2023, 1, 1), end=date(2024, 6, 1)
```
Comment (SypTitan) [inline loefsys/contacts/tests/test_models.py:152]:
```suggestion
            Membership, 
            member=self.member, 
            start=date(2023, 1, 1), 
            end=date(2024, 6, 1)
```

File: loefsys/contacts/tests/test_models.py [hunk 3]
```diff
@@ -42,42 +54,206 @@ def test_validation_person(self):
...
+            member=self.member,
+            start=date(2022, 1, 1),
+            end=date(2023, 12, 31),
+        )
+        self.assertTrue(validate_overlap(membership1, [membership2]))
+
+    def test_no_overlap_null_end(self):
+        """Test that there is no overlap when an ongoing membership has no end date."""
+        membership1 = G(
+            Membership, member=self.member, start=date(2024, 1, 1), end=None
```
Comment (SypTitan) [inline loefsys/contacts/tests/test_models.py:181]:
```suggestion
            Membership, 
            member=self.member, 
            start=date(2024, 1, 1), 
            end=None
```

File: loefsys/contacts/tests/test_models.py [hunk 4]
```diff
@@ -42,42 +54,206 @@ def test_validation_person(self):
...
+            member=self.member,
+            start=date(2023, 1, 1),
+            end=date(2023, 12, 31),
+        )
+        self.assertFalse(validate_overlap(membership1, [membership2]))
+
+    def test_overlap_null_end(self):
+        """Test that there is overlap with an ongoing membership."""
+        membership1 = G(
+            Membership, member=self.member, start=date(2023, 6, 1), end=None
```
Comment (SypTitan) [inline loefsys/contacts/tests/test_models.py:194]:
```suggestion
            Membership, 
            member=self.member, 
            start=date(2023, 6, 1), 
            end=None
```

File: loefsys/contacts/tests/test_models.py [hunk 5]
```diff
@@ -42,42 +54,206 @@ def test_validation_person(self):
...
+            member=self.member,
+            start=date(2020, 1, 1),
+            end=date(2020, 12, 31),
+        )
+        self.assertFalse(validate_overlap(membership1, [membership2, membership3]))
+
+    def test_overlap_multiple_memberships(self):
+        """Test that there is overlap with multiple overlapping memberships."""
+        membership1 = G(
+            Membership, member=self.member, start=date(2023, 6, 1), end=date(2024, 6, 1)
```
Comment (SypTitan) [inline loefsys/contacts/tests/test_models.py:229]:
```suggestion
            Membership, 
            member=self.member, 
            start=date(2023, 6, 1), 
            end=date(2024, 6, 1)
```

**Review by SypTitan — COMMENTED** [review]
Noticed some cosmetic inconsistencies, the actual tests look good

**Review by JortRoelofs — APPROVED** [review]
Thanks!

**Pull request merged**